### ISSUE TITLE
Accuracy, FP, FN

### ISSUE THREAD

**Issue by TommyTang930** [MaybeShewill-CV/lanenet-lane-detection#308]
您好，感谢大神做出的贡献 ~

请教一个问题：如何将训练好的模型在test数据集上测试，计算出accuracy, fp, fn ? 

我看了该项目的issues，也有很多人提到了。不过，没有成功的解决方法。请问您这边是如何计算的呢？麻烦大神指点一下，非常感谢 ~

**Comment by MaybeShewill-CV** [comment]
@TommyTang930 我是直接计算像素级别的这个指标 你可以参考repo里的代码:)

**Comment by TommyTang930** [comment]
@MaybeShewill-CV 
大神，我读了一下该项目下tools文件夹下的evaluate_model_utils.py里面的代码，比如：

def calculate_model_precision(input_tensor, label_tensor):
    """
    calculate accuracy acc = correct_nums / ground_truth_nums
    :param input_tensor: binary segmentation logits
    :param label_tensor: binary segmentation label
    :return:
    """
    logits = tf.nn.softmax(logits=input_tensor)
    final_output = tf.expand_dims(tf.argmax(logits, axis=-1), axis=-1)
    idx = tf.where(tf.equal(final_output, 1))
    pix_cls_ret = tf.gather_nd(label_tensor, idx)
    accuracy = tf.count_nonzero(pix_cls_ret)
    accuracy = tf.divide(
        accuracy,
        tf.cast(tf.shape(tf.gather_nd(label_tensor, tf.where(tf.equal(label_tensor, 1))))[0], tf.int64))
    return accuracy

请问大神，这个input_tensor是不是在一张test数据集中的图片，预测后的tensor？ 然后与test_label.json中的第一行lane计算，得到的就是accuracy，然后累加所有test中图片的计算结果，最后做一个均值，不知道这样理解对不对？ 很抱歉，新手上路，麻烦大神了。

**Comment by MaybeShewill-CV** [comment]
@TommyTang930 input tensor是inference result. label tensor就是你制作的ground truth label.你查看下代码就知道了:)

**Comment by TommyTang930** [comment]
@MaybeShewill-CV 
您好，

问题一：我在TuSimple的test数据集上跑通后，只是输出了车道线检测后的效果图，没有输出对应的预测json文件，请问如何添加代码生成该文件呢？

问题二：在该项目下新建了Log文件夹，在training过程中，没有看到输出的log文件，请问这是什么原因？是否需求配置代码中输出路劲呢？

**Comment by MaybeShewill-CV** [comment]
@TommyTang930 1.没有生成pred.json的代码需要自己对照tusimple benchmark的要求开发 2.在项目根目录下新建logs文件夹

**Comment by TommyTang930** [comment]
@MaybeShewill-CV 
好的，非常感谢 :)

**Comment by TommyTang930** [comment]
@MaybeShewill-CV 

您好，在写脚本文件，遇到问题如下：

问题描述

LaneNet训练完TuSimple的Training数据集后，用模型在TuSimple的test数据集上inference，获得了每一张test数据集上图片的车道线检测效果。通过查看源代码，可以获取检测后得到的每条车道线的坐标，比如：
source image :  ./data/tusimple_test_image/0.jpg
车道线数量 :  4
第1条车道线的shape :  (2522, 2)
第2条车道线的shape :  (2420, 2)
第3条车道线的shape :  (1560, 2)
第4条车道线的shape :  (1001, 2)
原图的shape :  (720, 1280, 3)

不过，我读取了一下TuSimple的test数据集的test_label.json文件，其格式是："lanes", "h_samples", "raw_file"，尤其是lanes字段，其包含的是每条车道线的横坐标，每条车道线的length固定是 56 ，比如：
[-2, -2, -2, -2, -2, -2, -2, 616, 591, 564, 538, 509, 480, 451, 422, 391, 358, 325, 292, 259, 227, 194, 161, 128, 95, 62, 29, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2, -2]

请问：如何将在test数据集上检测得到的车道线的横坐标改成这种格式呢？

非常感谢 ～
